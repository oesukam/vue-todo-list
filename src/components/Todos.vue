<template>
  <section class="container todos-container">
    <div class="row align-items-center">
      <div class="col-sm-6 offset-md-3">
        <div class="todos-box">
          <h1 class="title has-text-centered">My Todos</h1>
          <div class="input-group mb-5">
            <input
              class="form-control"
              placeholder="Enter your todo"
              name="text"
              v-model="text"
              v-on:keydown.enter="addNewTodo"
            >
            <div class="input-group-append">
              <button
                v-on:click="addNewTodo"
                class="btn btn-outline-secondary"
                type="button"
                id="add-btn"
              >Add</button>
            </div>
          </div>
          <div class="my-todos">
            <SingleTodo
              v-for="(todo, index) in todos"
              :index="index"
              :key="todo.id"
              :todo="todo"
              v-on:removeTodo="removeTodo"
              v-on:checkTodo="checkTodo"
            />
          </div>
        </div>
      </div>
    </div>
  </section>
</template>

<script>
import { mapGetters, mapActions } from 'vuex';
import uuid from 'uuid';
import SingleTodo from './SingleTodo';

export default {
  name: 'Todos',
  components: {
    SingleTodo
  },
  data() {
    return {
      text: ''
    };
  },
  computed: {
    ...mapGetters({
      todos: 'getTodos'
    })
  },
  methods: {
    addNewTodo() {
      const { text } = this;
      if (text) {
        const todo = {
          id: uuid.v4(),
          text,
          done: false,
          deleting: false
        };
        this.addTodo(todo).then(() => {
          this.text = '';
        });
      }
    },
    ...mapActions(['addTodo', 'checkTodo', 'removeTodo'])
  }
};
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped>
.todos-box {
  min-height: 300px;
  box-shadow: 0px 0px 2px 1px rgba(0, 0, 0, 0.2);
  padding: 2rem;
  border-radius: 0.5rem;
}
.todos-box h1 {
  margin-bottom: 2rem;
}
</style>
