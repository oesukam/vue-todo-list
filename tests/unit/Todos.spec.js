import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import Todos from '@/components/Todos';

const localVue = createLocalVue();

localVue.use(Vuex);
describe('Todos.vue', () => {
  let store;
  let actions;
  let getters;
  beforeEach(() => {
    actions = {
      addTodo: jest.fn(),
      checkTodo: jest.fn(),
      removeTodo: jest.fn()
    };
    getters = {
      getTodos: jest.fn(() => [{ id: '2', text: '' }])
    };
    store = new Vuex.Store({
      actions,
      getters
    });
  });

  test('renders correctly', () => {
    const wrapper = shallowMount(Todos, {
      store,
      localVue
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  describe('when clicking on `add` todo', () => {
    it('should call `addTodo` without adding', () => {
      const wrapper = shallowMount(Todos, {
        store,
        localVue
      });
      wrapper.find('#add-btn').trigger('click');
      expect(wrapper.vm.text).toBe('');
    });

    it('should call `addTodo` and add new todo`', () => {
      const wrapper = shallowMount(Todos, {
        store,
        localVue
      });
      wrapper.setData({
        text: 'text'
      });
      wrapper.find('#add-btn').trigger('click');
      expect(actions.addTodo).toHaveBeenCalled();
    });
  });
});
