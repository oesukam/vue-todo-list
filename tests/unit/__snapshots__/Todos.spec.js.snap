// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todos.vue renders correctly 1`] = `
<section
  class="container todos-container"
>
  <div
    class="row align-items-center"
  >
    <div
      class="col-sm-6 offset-md-3"
    >
      <div
        class="todos-box"
      >
        <h1
          class="title has-text-centered"
        >
          My Todos
        </h1>
         
        <div
          class="input-group mb-5"
        >
          <input
            class="form-control"
            name="text"
            placeholder="Enter your todo"
          />
           
          <div
            class="input-group-append"
          >
            <button
              class="btn btn-outline-secondary"
              id="add-btn"
              type="button"
            >
              Add
            </button>
          </div>
        </div>
         
        <div
          class="my-todos"
        >
          <singletodo-stub
            index="0"
            todo="[object Object]"
          />
        </div>
      </div>
    </div>
  </div>
</section>
`;
